src/main/java/ro/hiringsystem/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,206 @@
+package ro.hiringsystem.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import ro.hiringsystem.service.UserMapperService;
+import ro.hiringsystem.model.auxiliary.AcademicExperience;
+import ro.hiringsystem.model.auxiliary.CV;
+import ro.hiringsystem.model.auxiliary.Project;
+import ro.hiringsystem.model.auxiliary.WorkExperience;
+import ro.hiringsystem.model.dto.CandidateUserDto;
+import ro.hiringsystem.model.dto.UserDto;
+import ro.hiringsystem.security.JwtService;
+import ro.hiringsystem.security.auth.AuthenticationRequest;
+import ro.hiringsystem.security.auth.AuthenticationResponse;
+import ro.hiringsystem.security.auth.RegisterRequest;
+import ro.hiringsystem.security.token.Token;
+import ro.hiringsystem.security.token.TokenRepository;
+import ro.hiringsystem.security.token.TokenType;
+import ro.hiringsystem.service.AuthenticationService;
+import ro.hiringsystem.service.UserService;
+
+import java.io.IOException;
+import java.net.URL;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private final PasswordEncoder passwordEncoder;
+    private final JwtService jwtService;
+    private final AuthenticationManager authenticationManager;
+
+    private final TokenRepository tokenRepository;
+
+    private final UserService<UserDto> userService;
+
+    private final UserMapperService userMapper;
+
+    @Override
+    public AuthenticationResponse register(RegisterRequest request) {
+
+        //code to generate user required here
+        //using a manual user for testing purposes
+        //taken from CandidateUsersController
+
+        try {
+            CandidateUserDto candidateUser = CandidateUserDto.builder()
+                    .id(UUID.randomUUID())
+                    .firstName(request.getFirstName())
+                    .lastName(request.getLastName())
+                    .password(passwordEncoder.encode(request.getPassword()))
+                    .primaryEmail(request.getEmail())
+                    .mailList(List.of())
```

Comment:
Since I also asked for the MEMBER OF modification, please add the mail inside List.of() as well